server/api/src/rsDataObjTrim.cpp

```diff
@@ -12,172 +12,252 @@
+        rstrcpy(info.dataMode, obj.mode().c_str(), sizeof(info.dataMode));
+        rstrcpy(info.dataComments, obj.r_comment().c_str(), sizeof(info.dataComments));
+        rstrcpy(info.dataCreate, obj.create_ts().c_str(), sizeof(info.dataCreate));
+        rstrcpy(info.dataModify, obj.modify_ts().c_str(), sizeof(info.dataModify));
+        rstrcpy(info.rescHier, obj.resc_hier().c_str(), sizeof(info.rescHier));
+
+        return info;
     }
-    else if ( remoteFlag == REMOTE_HOST ) {
-        status = rcDataObjTrim( rodsServerHost->conn, dataObjInp );
-        return status;
+
+    int get_time_of_expiration(const char* age_kw) {
+        if (!age_kw) {
+            return 0;
+        }
+        try {
+            const auto age_in_minutes = std::atoi(age_kw);
+            if ( age_in_minutes > 0 ) {
+                const auto now = time(0);
+                const auto age_in_seconds = age_in_minutes * 60;
+                return now - age_in_seconds;
+            }
+            return 0;
+        }
+        catch (const std::exception&) {
+            return 0;
+        }
     }
 
-    // =-=-=-=-=-=-=-
-    // working on the "home zone", determine if we need to redirect to a different
-    // server in this zone for this operation.  if there is a RESC_HIER_STR_KW then
-    // we know that the redirection decision has already been made
-    std::string       hier;
-    int               local = LOCAL_HOST;
-    rodsServerHost_t* host  =  0;
-    char* hier_char = getValByKey( &dataObjInp->condInput, RESC_HIER_STR_KW );
-    if ( hier_char == NULL ) {
-        // set a repl keyword here so resources can respond accordingly
-        addKeyVal( &dataObjInp->condInput, IN_REPL_KW, "" );
-        irods::error ret = irods::resource_redirect( irods::UNLINK_OPERATION, rsComm,
-                           dataObjInp, hier, host, local );
-        if ( !ret.ok() ) {
-            std::stringstream msg;
-            msg << __FUNCTION__;
-            msg << " :: failed in irods::resource_redirect for [";
-            msg << dataObjInp->objPath << "]";
-            irods::log( PASSMSG( msg.str(), ret ) );
-            return ret.code();
-        }
-        // =-=-=-=-=-=-=-
-        // we resolved the redirect and have a host, set the hier str for subsequent
-        // api calls, etc.
-        addKeyVal( &dataObjInp->condInput, RESC_HIER_STR_KW, hier.c_str() );
-
-    } // if keyword
-
-    if ( getValByKey( &dataObjInp->condInput, ADMIN_KW ) != NULL ) {
-        if ( rsComm->clientUser.authInfo.authFlag < LOCAL_PRIV_USER_AUTH ) {
-            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
-        }
-        accessPerm = NULL;
+    unsigned long get_minimum_replica_count(const char* copies_kw) {
+        if (!copies_kw) {
+            return DEF_MIN_COPY_CNT;
+        }
+        try {
+            const auto minimum_replica_count = std::stoul(copies_kw);
+            if (minimum_replica_count <= 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Checking if unsigned variable 'minimum_replica_count' is less than zero.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4331469)